lego/connections.py

```diff
@@ -87,4 +89,51 @@ def close(self) -> None:
         self._machine.close()
 
 
+class RPyCConnection(BaseConnection):
+    """RPyC wrapper for component connection.
+
+    In case the machine doesn't already run SlaveService, we will try to use SSH to upload and
+    deploy RPyC SlaveService.
+    """
+
+    def __init__(self, hostname: str, username: Optional[str], password: Optional[str]) -> None:
+        """Connects (or start with SSH if needed) to RPyC remote SlaveService.
+
+        Connect to RPyC SlaveService on remote machine. If the service isn't running, try
+        to deploy it with SSHConnection and RPyC zero deploy library.
+
+        Args:
+            hostname: The hostname of the component we want to connect to.
+            username: Username for SSH login (if needed).
+            password: Password for SSH login (if needed).
+        """
+
+        self._server = None
+        try:
+            # Checks if the machine already runs RPyC SlaveService.
+            self._connection = rpyc.classic.connect(hostname, keepalive=True)
+        except ConnectionRefusedError:
+            if username is None or password is None:
+                # Not given necessary SSH credentials.
+                raise
+
+            with SSHConnection(hostname, username, password) as ssh:
+                # Upload RPyC and start SlaveService in a temporarily directory.
+                self._server = DeployedServer(ssh)
+                self._connection = self._server.classic_connect()
```

Comment:
```suggestion
                with DeployedServer(ssh) as server:
                    server.classic_connect()
```

Reply:
Deployed server should be opened as long as rpyc connection is open.
In your suggested solution, after DeployedServer is closed you can't use rpyc connection any more.